nifi-nar-bundles/nifi-prometheus-bundle/nifi-prometheus-nar/src/main/resources/META-INF/NOTICE

```diff
@@ -0,0 +1,44 @@
+Copyright 2015-2017 The Apache Software Foundation
```

Comment:
This NOTICE file seems to need quite a bit of work. 

Dependency artifacts I do not see covered in this NOTICE (in addition, please make sure that anything in this notice is also included in the nifi-assembly notice, as this NAR will be included in the nifi-assembly binary as well):

- `io.prometheus:simpleclient`
- `io.prometheus:simpleclient_hotspot`
- `io.prometheus:simpleclient_servlet`
- `org.eclipse.jetty` (multiple artifacts)
- Java Servlet API

In addition, it references artifacts I do not see as maven dependencies (though they could be transitively brought in... I haven't checked)

Reply:
I used **maven-project-info-reports-plugin** and was able to extract  below details.  How can this information be used to fill the NOTICE file? 

// ------------------------------------------------------------------
// Transitive dependencies of this project determined from the
// maven pom organized by organization.
// ------------------------------------------------------------------

nifi-prometheus-reporting-task


From: 'an unknown organization'
  - Metrics Core Library (http://metrics.codahale.com/metrics-core/) com.yammer.metrics:metrics-core:jar:2.2.0
    License: Apache License 2.0  (http://www.apache.org/licenses/LICENSE-2.0.html)
  - jBCrypt (http://www.mindrot.org/projects/jBCrypt) de.svenkubiak:jBCrypt:bundle:0.4.1
    License: ISC/BSD licence  (http://www.mindrot.org/files/jBCrypt/LICENSE)
  - Prometheus Java Simpleclient (http://github.com/prometheus/client_java/simpleclient) io.prometheus:simpleclient:bundle:0.3.0
    License: The Apache Software License, Version 2.0  (http://www.apache.org/licenses/LICENSE-2.0.txt)
  - Prometheus Java Simpleclient Common (http://github.com/prometheus/client_java/simpleclient_common) io.prometheus:simpleclient_common:bundle:0.3.0
    License: The Apache Software License, Version 2.0  (http://www.apache.org/licenses/LICENSE-2.0.txt)
  - Prometheus Java Simpleclient Hotspot (http://github.com/prometheus/client_java/simpleclient_hotspot) io.prometheus:simpleclient_hotspot:bundle:0.3.0
    License: The Apache Software License, Version 2.0  (http://www.apache.org/licenses/LICENSE-2.0.txt)
  - Prometheus Java Simpleclient Servlet (http://github.com/prometheus/client_java/simpleclient_servlet) io.prometheus:simpleclient_servlet:bundle:0.3.0
    License: The Apache Software License, Version 2.0  (http://www.apache.org/licenses/LICENSE-2.0.txt)
  - Bouncy Castle PKIX, CMS, EAC, TSP, PKCS, OCSP, CMP, and CRMF APIs (http://www.bouncycastle.org/java.html) org.bouncycastle:bcpkix-jdk15on:jar:1.60
    License: Bouncy Castle Licence  (http://www.bouncycastle.org/licence.html)
  - Bouncy Castle Provider (http://www.bouncycastle.org/java.html) org.bouncycastle:bcprov-jdk15on:jar:1.60
    License: Bouncy Castle Licence  (http://www.bouncycastle.org/licence.html)

From: 'Apache NiFi Project' (http://nifi.apache.org/)
  - nifi-security-utils (http://nifi.apache.org/nifi-commons/nifi-security-utils) org.apache.nifi:nifi-security-utils:jar:1.9.0
    License: Apache License, Version 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
  - nifi-ssl-context-service (http://nifi.apache.org/nifi-nar-bundles/nifi-standard-services/nifi-ssl-context-bundle/nifi-ssl-context-service) org.apache.nifi:nifi-ssl-context-service:jar:1.9.0
    License: Apache License, Version 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
  - nifi-standard-services-api-nar (http://nifi.apache.org/nifi-nar-bundles/nifi-standard-services/nifi-standard-services-api-nar) org.apache.nifi:nifi-standard-services-api-nar:nar:1.9.0
    License: Apache License, Version 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
  - nifi-utils (http://nifi.apache.org/nifi-commons/nifi-utils) org.apache.nifi:nifi-utils:jar:1.9.0
    License: Apache License, Version 2.0  (http://www.apache.org/licenses/LICENSE-2.0)

From: 'The Apache Software Foundation' (http://www.apache.org/)
  - Apache Commons Logging (http://commons.apache.org/proper/commons-logging/) commons-logging:commons-logging:jar:1.2
    License: The Apache Software License, Version 2.0  (http://www.apache.org/licenses/LICENSE-2.0.txt)
  - Apache HttpClient (http://hc.apache.org/httpcomponents-client) org.apache.httpcomponents:httpclient:jar:4.5.3
    License: Apache License, Version 2.0  (http://www.apache.org/licenses/LICENSE-2.0.txt)
  - Apache HttpCore (http://hc.apache.org/httpcomponents-core-ga) org.apache.httpcomponents:httpcore:jar:4.4.6
    License: Apache License, Version 2.0  (http://www.apache.org/licenses/LICENSE-2.0.txt)

From: 'The Apache Software Foundation' (https://www.apache.org/)
  - Apache Commons Codec (http://commons.apache.org/proper/commons-codec/) commons-codec:commons-codec:jar:1.11
    License: Apache License, Version 2.0  (https://www.apache.org/licenses/LICENSE-2.0.txt)
  - Apache Commons Lang (http://commons.apache.org/proper/commons-lang/) org.apache.commons:commons-lang3:jar:3.8.1
    License: Apache License, Version 2.0  (https://www.apache.org/licenses/LICENSE-2.0.txt)

From: 'Webtide' (http://webtide.com)
  - Jetty :: Utilities (http://www.eclipse.org/jetty) org.eclipse.jetty:jetty-util:jar:9.3.8.v20160314
    License: Apache Software License - Version 2.0  (http://www.apache.org/licenses/LICENSE-2.0)    License: Eclipse Public License - Version 1.0  (http://www.eclipse.org/org/documents/epl-v10.php)

Reply:
What I do is use `mvn dependency:tree | grep compile` on the NAR project and hunt through existing NOTICE and LICENSE files for existing verbiage that can be applied. Virtually everything you have there is already in an existing NOTICE or LICENSE file. For the NiFi ones, you can skip references to those. Everything else, I'd recommend you copy and paste from existing files. As far as I can tell, the organization for Prometheus dependencies would be the Linux Foundation since that is what their site asserts viz a viz the copyright notice.